src/main/java/cuchaz/enigma/EnigmaProject.java

```diff
@@ -39,13 +41,16 @@
 
 	private final ClassCache classCache;
 	private final JarIndex jarIndex;
+	private final byte[] jarChecksum;
 
 	private EntryRemapper mapper;
 
-	public EnigmaProject(Enigma enigma, ClassCache classCache, JarIndex jarIndex) {
+	public EnigmaProject(Enigma enigma, ClassCache classCache, JarIndex jarIndex, byte[] jarChecksum) {
+		Preconditions.checkArgument(jarChecksum.length == EnigmaServer.CHECKSUM_SIZE);
```

Comment:
Any reason to do this?

Reply:
If it's the wrong length then the login packet will write a different number of bytes than the server reads.